kubeflow-pipelines.yaml

```diff
@@ -157,13 +148,13 @@ subpackages:
       - runs: |
           mkdir -p ${{targets.subpkgdir}}/server
           mkdir -p ${{targets.subpkgdir}}/client
+          mkdir -p ${{targets.subpkgdir}}/usr/local/lib
           cd frontend/
-          npm ci
-          npm run postinstall
+          npm i
           npm run build
           cp -r ./build/* ${{targets.subpkgdir}}/client/
           cd ./server
-          npm ci
+          npm i
           npm run build
           mv ./* ${{targets.subpkgdir}}/server/
 
```

Comment:
The npm modules that have CVEs are all copied through the `/server` dir, because `/build` doesn't contain any.
The build steps (copied from their dockerfile) actually ship all dev dependencies with `/server`.
You can fix that by `npm ci` to build and then `rm` + `npm ci --production` to install only runtime deps.
Combine it with an extra `npm audit fix` before and you _should_ be dealing with a much more manageable list of CVEs.

```suggestion
          npm ci
          npm run postinstall
          npm run build
          cp -r ./build/* ${{targets.subpkgdir}}/client/
          cd ./server
          npm ci
          npm run build
          rm -rf ./node_modules
          npm audit fix --package-lock-only
          npm ci --production
          mv ./* ${{targets.subpkgdir}}/server/

```

Note: I was able to get the frontend running with `npm audit fix --package-lock-only --force`, but I have no idea what will break at runtime if you do this, so you'd want to do some thorough testing or resort to advisories for these.

Reply:
thank you so much!

>but I have no idea what will break at runtime if you do this, so you'd want to do some thorough testing or resort to advisories for these.

yes definitely I hope the tests is going to tell us what went wrong.

Reply:
Is there still more to do with this review comment?  I re-ran the failing scan CI check and there's still some CVEs detected which could be patched or advisories created for.

Reply:
I already created here: 

- https://github.com/wolfi-dev/advisories/pull/432